### PR TITLE
PPNA-3887 Analytics annotations

### PR DESCRIPTION
**Pull request by pulina**
Connected with https://github.com/PiwikPRO/PPMS-Analytics-Reporting/pull/1804

### CODE REVIEW

**Comment by JacekClearcode** [comment]
For viewing pleasure https://developers.piwik.pro/en/feature-ppna-3887-annotation-api/custom_reports/object_management_api/object_management_api.html#tag/User-Annotations

File: custom_reports/object_management_api/paths/annotation_system.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    Returns a list all System Annotations available for a website.
+    Please note that each request must be authenticated.
+  parameters:
+      # - test:
+      #   in: query
+      #   description: ID of the website.
+      #   required: true
+      #   schema:
+      #     type: string
+      - $ref: '#/parameters/WebsiteId'
```
Comment (mrfijas) [inline custom_reports/object_management_api/paths/annotation_system.yaml:15]: system annotations don't use website_id parameter

File: custom_reports/object_management_api/paths/annotation_system.yaml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+          count:
+            type: integer
+            minimum: 0
+            description: Number of total annotations matching query criteria.
+            example: 1
+          results:
+            type: array
+            items:
+              $ref: '#/definitions/AnnotationSystem'
+    '403':
```
Comment (mrfijas) [inline custom_reports/object_management_api/paths/annotation_system.yaml:38]: 400 is also possible here (i.e. unparesable `date_from`)

File: custom_reports/object_management_api/definitions/annotation_user.yaml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+      - website_id
+      - date
+    properties:
+      visibility:
+        enum:
+          - private
+          - public
+        default: private
+        description: |
+          Determines whether the attribution should be visible only to the author or to everyone with access to the website.
```
[the remark this answers is not in the document]
Comment (pulina) [reply custom_reports/object_management_api/definitions/annotation_user.yaml:16]: Copped from also wrong internal api docs https://github.com/PiwikPRO/PPMS-Analytics-Reporting/blob/dev/docs/_common.yaml#L47

File: custom_reports/object_management_api/definitions/annotation.yaml [hunk 4]
```diff
@@ -0,0 +1,10 @@
+type: object
```
Comment (pulina) [inline custom_reports/object_management_api/definitions/annotation.yaml:1]: As suggested @vanzi we should consider change response from blank object to object called "data" including actual response. Src: https://piwikpro.atlassian.net/wiki/spaces/PPMS/pages/7556169719/Public+API+Design+response+format+naming+urls

File: custom_reports/object_management_api/paths/annotation_user_detail.yaml [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+          schema:
+            type: object
+            required:
+              - data
+            properties:
+              data:
+                $ref: '#/definitions/AnnotationUserDetails'
+    '403':
+      description: Forbidden
+put:
```
Comment (mrfijas) [inline custom_reports/object_management_api/paths/annotation_user_detail.yaml:25]: I think `patch` is more user friendly (and we've got it in internal docs)

**Pull request merged**